src/Linux/Native/RunAsUser/src/main.cpp

```diff
@@ -49,8 +50,10 @@ static constexpr char RTVS_RESPONSE_TYPE_RTVS_ERROR[] = "rtvs-error";
 
 static constexpr char RTVS_MSG_AUTH_ONLY[] = "AuthOnly";
 static constexpr char RTVS_MSG_AUTH_AND_RUN[] = "AuthAndRun";
+static constexpr char RTVS_MSG_KILL_PROCESS[] = "KillProcess";
 
 static constexpr char RTVS_RHOST_PATH[] = "/usr/lib/rtvs/Microsoft.R.Host";
+static constexpr char RTVS_KILL_PATH[] = "/bin/kill";
```

Comment:
Sorry, I should have clarified what I meant here.

If we pass the binary as an argument in the JSON message, instead of hardcoding, then this thing doesn't need to know what it is starting. And then you can pass the host binary or kill or whatever from the managed side, and there's no special code for kill.

(And if this doesn't work for some reason, then there's no point using kill as a process instead of the corresponding API.)